src/client/src/apps/SimpleLauncher/spotlight/components/TradeExecutionOverlay.tsx

```diff
@@ -0,0 +1,210 @@
+import numeral from 'numeral'
+import React, { FC, useEffect, useMemo, useState } from 'react'
+import {
+  TradeExecutionActionContainer,
+  TradeExecutionContainer,
+  TradeResponseContainer,
+} from './styles'
+
+import { useTradeExecution } from './useTradeExecution'
+import { useMarketService } from './useMarketService'
+import { IndeterminateLoadingBar, IndeterminateLoadingBarStatus } from './IndeterminateLoadingBar'
+import { DetectIntentResponse } from 'dialogflow'
+import { getTradeRequest } from 'rt-interop'
+import { TradeSuccessResponse } from 'apps/MainRoute/widgets/spotTile/model/executeTradeRequest'
+import { Direction } from 'rt-types'
+
+type TradeExecutionRequestStatus = 'incomplete' | 'complete'
+export type TradeExecutionStatus = 'waiting' | 'executing' | 'success' | 'failure' | undefined
+
+export type TradeExecutionProps = {
+  handleReset: () => void
+  handleClearSearchInput: () => void
+  response: DetectIntentResponse
+}
+
+export const TradeExecutionOverlay: FC<TradeExecutionProps> = ({
+  response,
+  handleReset,
+  handleClearSearchInput,
+}) => {
+  const [requestStatus, setRequestStatus] = useState<TradeExecutionRequestStatus>('incomplete')
+  const [tradeStatus, setTradeStatus] = useState<TradeExecutionStatus>()
+  const [loadingBarStatus, setLoadingBarStatus] = useState<IndeterminateLoadingBarStatus>()
```

Comment:
I think this is inherited from the existing code, and maybe not for now, but it seems like you could derive loadingBarStatus from the other pieces of state rather than having to explicitly set it.